From db46e2802c135720b512c5151a60594cf8c767ac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Pereira?= <luis.artur.pereira@gmail.com>
Date: Mon, 2 Feb 2015 11:42:43 -0800
Subject: [PATCH 10/15] QIconLoader: enable an easy case of transactional
 processing

It's easy to do the work on the side and then commit. This is strongly
exception safe, but in Qt, we don't care. But transactional code, when
this easy to achieve, is also clearer.

Port of commit
https://qt.gitorious.org/qt/qtbase/commits/de068472f1368450bad4aff74eec0bc112ae413a
---
 qiconfix/qiconloader.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/qiconfix/qiconloader.cpp b/qiconfix/qiconloader.cpp
index 8ae43ee..2c3502a 100644
--- a/qiconfix/qiconloader.cpp
+++ b/qiconfix/qiconloader.cpp
@@ -666,15 +666,16 @@ void QIconLoaderEngineFixed::virtual_hook(int id, void *data)
         {
             QIconEngine::AvailableSizesArgument &arg
                     = *reinterpret_cast<QIconEngine::AvailableSizesArgument*>(data);
-            arg.sizes.clear();
             const int N = m_entries.size();
-            arg.sizes.reserve(N);
+            QList<QSize> sizes;
+            sizes.reserve(N);
 
             // Gets all sizes from the DirectoryInfo entries
             for (int i = 0; i < N; ++i) {
                 int size = m_entries.at(i)->dir.size;
-                arg.sizes.append(QSize(size, size));
+                sizes.append(QSize(size, size));
             }
+            arg.sizes.swap(sizes); // commit
         }
         break;
     case QIconEngine::IconNameHook:
-- 
1.9.0

